src/wasm/kotlin/src/arcs/Entity.kt

```diff
@@ -96,18 +81,49 @@ class StringDecoder(private var str: String) {
 class StringEncoder(private val sb: StringBuilder = StringBuilder()) {
 
     companion object {
-      fun encodeDictionary(dict: Map<String, String?>): String {
-        val sb = StringBuilder()
-        sb.append(dict.size).append(":")
+        fun encodeDictionary(dict: Map<String, Any?>): String {
+            val sb = StringBuilder()
+            sb.append(dict.size).append(":")
 
-        for((key, value) in dict) {
-            value?.let {
+            for((key, value) in dict) {
                 sb.append(key.length).append(":").append(key)
-                sb.append(it.length).append(":").append(it)
+                sb.append(encodeValue(value))
+            }
+            return sb.toString()
+        }
+
+        fun encodeList(list: List<Any>): String {
+          val sb = StringBuilder()
+          sb.append(list.size).append(":")
+          
+          for(value in list) {
+            sb.append(encodeValue(value))
+          }
+          return sb.toString()
+        }
+
+        fun encodeValue(value: Any?): String {
```

Comment:
Probably don't need explicit stringbuilder usage for this method either... consider using string interpolation:

```kotlin
return when (value) {
    is String -> "T${value.length}:$value"
    is Map<*, *> -> {
        @Suppress("UNCHECKED_CAST")
        val dictString = encodeDictionary(value as Map<String, Any>)
        "D${dictString.length}:$dictString"
    }
    is List<*> -> {
        @Suppress("UNCHECKED_CAST")
        val listString = encodeList(value as List<Any>)
        "A${listString.length}:$listString"
    }
    else -> throw IllegalArgumentException("Unknown expression.")
}
```